src/openrct2/ride/Ride.cpp

```diff
@@ -504,7 +504,7 @@ bool TrackBlockGetNextFromZero(
 {
     auto trackPos = startPos;
 
-    if (!(direction_start & TRACK_BLOCK_2))
+    if (!TrackPieceDirectionIsDiagonal(direction_start))
```

Comment:
Use this for everything - especially since `&`ing with `kNumOrthogonalDirections` doesn’t make too much logical sense.